### PR TITLE
feat(tempest): Endpoint for fetching tempest credentials

### PR DESCRIPTION
**Pull request by vgrozdanic**
Endpoint only for GET request. Responsible for fetching all tempest credentials for a given project.

Part of https://github.com/getsentry/team-gdx/issues/52

2 more endpoints will be done as a part of separate PRs to reduce the scope of the PR:
- POST - creation of new credentials record
- DELETE - deletion of the existing credentials record

### CODE REVIEW

**Review by obostjancic — APPROVED** [review]

**Review by tobias-wilfert — APPROVED** [review]

File: src/sentry/api/urls.py [hunk 1]
```diff
@@ -3265,6 +3266,12 @@ def create_group_urls(name_prefix: str) -> list[URLPattern | URLResolver]:
         SecretScanningGitHubEndpoint.as_view(),
         name="sentry-api-0-secret-scanning-github",
     ),
+    # Tempest
+    re_path(
+        r"^(?P<organization_id_or_slug>[^\/]+)/(?P<project_id_or_slug>[^\/]+)/tempest-credentials/$",
+        TempestCredentialsEndpoint.as_view(),
+        name="sentry-api-0-project-tempest-credentials",
+    ),
```
Comment (shellmayr) [inline src/sentry/api/urls.py:3274]: As this is a ProjectEndpoint, we should probably put this routing rule into the `PROJECT_URLS` constant in this file to place it with the other project endpoints, or are we not doing this on purpose?

Comment (vgrozdanic) [reply src/sentry/api/urls.py:3274]: Uuups, my bad, wanted to add it to the end of the list, but by accident added it to the end of the file 😬

File: tests/sentry/tempest/endpoints/test_tempest_credentials.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class TestTempestCredentials(APITestCase):
+    endpoint = "sentry-api-0-project-tempest-credentials"
+
+    def test_create_tempest_credentials(self):
+        with Feature({"organizations:tempest-access": True}):
+            credentials = [self.create_tempest_credentials(self.project) for _ in range(5)]
+
+            other_project = self.create_project(organization=self.organization)
+            # credentials connected to other project which should not be included in the response
+            [self.create_tempest_credentials(other_project) for _ in range(5)]
```
Comment (shellmayr) [inline tests/sentry/tempest/endpoints/test_tempest_credentials.py:14]: What is the purpose of this list comprehension that isn't persisted in any way?

Comment (vgrozdanic) [reply tests/sentry/tempest/endpoints/test_tempest_credentials.py]: no purpose, leftover of some code cleanup. "refactored" now

**Review by shellmayr — APPROVED** [review]

**Pull request merged**